lib/network/kmsAWS/Client.ts

```diff
@@ -101,20 +114,27 @@ export default class Client {
                 return;
             }
             logger.debug("AWS KMS: master encryption key created", { KeyMetadata: data?.KeyMetadata });
-            cb(null, data?.KeyMetadata?.KeyId);
+            const keyId = this.noAwsArn ? data?.KeyMetadata?.KeyId : data?.KeyMetadata?.Arn ?? data?.KeyMetadata?.KeyId;
```

Comment:
This looks quite cryptic, could you spread it on multiple lines with explicit `if`s instead?

Reply:
```javascript
            const keyMd = data?.KeyMetadata;
            const keyId = this.noAwsArn
                ? keyMd?.KeyId
                : keyMd?.Arn || keyMd?.KeyId;

```
Does this looks better ?

Reply:
Fixed https://github.com/scality/Arsenal/pull/2349/files#diff-fb4f257ae2f25bd280160e3cdec22c57fdd9aaa0ee35637e056146ce2b35e6ecR124-R131